internal/fuzzbinder/fuzzclaim.go

```diff
@@ -180,3 +181,122 @@ func parseFuzzClaimBytes(statementBytes []byte) (*intoto.Statement, error) {
+		projectFuzzEffort.FuzzTimeSeconds += fuzzEffort.FuzzTimeSeconds
+		projectFuzzEffort.NumberFuzzTests += fuzzEffort.NumberFuzzTests
+	}
+	// Generate fuzzing claim specification.
+	fuzzClaimSpec.PerProject = &FuzzStats{
+		BranchCoverage:  projectCoverage.BranchCoverage,
+		LineCoverage:    projectCoverage.LineCoverage,
+		DetectedCrashes: projectCrashes.Detected,
+		FuzzTimeSeconds: projectFuzzEffort.FuzzTimeSeconds,
+		NumberFuzzTests: projectFuzzEffort.NumberFuzzTests,
+	}
+	for _, fuzzTagret := range fuzzTargets {
+		targetStats := FuzzStats{
+			BranchCoverage:  fuzzersCoverage[fuzzTagret].BranchCoverage,
+			LineCoverage:    fuzzersCoverage[fuzzTagret].LineCoverage,
+			DetectedCrashes: fuzzersCrashes[fuzzTagret].Detected,
+			FuzzTimeSeconds: fuzzersFuzzEffort[fuzzTagret].FuzzTimeSeconds,
+			NumberFuzzTests: fuzzersFuzzEffort[fuzzTagret].NumberFuzzTests,
+		}
+		targetSpec := FuzzSpecPerTarget{
+			Name:      fmt.Sprintf("%s_%s_%s", fuzzParameters.FuzzEngine, fuzzParameters.ProjectName, fuzzTagret),
+			Path:      fmt.Sprintf("%s/fuzz/fuzz_targets/%s.rs", fuzzParameters.ProjectName, fuzzTagret),
+			FuzzStats: &targetStats,
+		}
+		fuzzClaimSpec.PerTarget = append(fuzzClaimSpec.PerTarget, targetSpec)
+	}
+	return &fuzzClaimSpec, nil
+}
+
+// GenerateFuzzClaim generates a fuzzing claim (an instance of intoto.Statement,
+// with AmberClaimV1 as the PredicateType and FuzzClaimV1 as the ClaimType) using the
+// fuzzing reports of OSS-Fuzz.
+func GenerateFuzzClaim(date string, fuzzParameters *FuzzParameters) (*intoto.Statement, error) {
+	var statement intoto.Statement
+	var predicate amber.ClaimPredicate
+	revisionDigest, err := GetCoverageRevision(date, fuzzParameters)
+	if err != nil {
+		return nil, err
+	}
+	fuzzTargets, err := GetFuzzTargets(date, fuzzParameters)
+	if err != nil {
+		return nil, err
+	}
+	// Generate Amber predicate
+	predicate.ClaimType = FuzzClaimV1
+	currentTime := time.Now()
+	tomorrow := time.Now().AddDate(0, 0, 1)
+	nextWeek := time.Now().AddDate(0, 0, 7)
+	predicate.IssuedOn = &currentTime
+	predicate.Validity = &amber.ClaimValidity{
+		NotBefore: &tomorrow,
+		NotAfter:  &nextWeek,
+	}
+	fuzzClaimSpec, err := generateFuzzClaimSpec(date, revisionDigest, fuzzParameters, fuzzTargets)
+	if err != nil {
+		return nil, err
+	}
+	predicate.ClaimSpec = *fuzzClaimSpec
+	evidences, err := GetEvidences(date, fuzzTargets, fuzzParameters)
+	if err != nil {
+		return nil, err
+	}
+	predicate.Evidence = evidences
+	// Generate intoto statement
+	statement.Type = intoto.StatementInTotoV01
+	subject := intoto.Subject{
+		Name:   fuzzParameters.ProjectHomepage,
+		Digest: revisionDigest,
+	}
+	statement.Subject = append(statement.Subject, subject)
+	statement.PredicateType = amber.AmberClaimV1
+	validFuzzPredicate, err := ValidateFuzzClaim(predicate)
+	if err != nil {
+		return nil, err
+	}
+	statement.Predicate = *validFuzzPredicate
+	validAmberPredicate, err := amber.ValidateAmberClaim(statement)
```

Comment:
This is strange. Are `ValidateFuzzClaim` and `ValidateAmberClaim` really needed? I think `ValidateFuzzClaim` subsumes `ValidateAmberClaim`. If they are both actually needed could you please add a comment explaining why?

Reply:
I added `ValidateAmberClaim` to `ValidateFuzzClaim` so that we don't have to call both. 

They are not really necessary since the generated claim should be valid by design, but it helps to check if there are problems (so it's just there as an extra precaution). I can remove them though.